api/permissions/permissions_calculator.py

```diff
@@ -107,11 +107,26 @@ def permissions(self) -> typing.Set[str]:
         ).union(
             reduce(
                 lambda a, b: a.union(b),
-                [role.permissions for role in self.roles],
+                [
+                    role_permission.permissions
+                    for role_permission in self.roles
+                    if not role_permission.role.tags
```

Comment:
Are we prefetching this data?

Reply:
None of these objects are ORM; they are dataclasses. We do prefetch/select related in RBAC when we build this